From 15c218c57ac4d5a0e44ec315886f119274a6d63d Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Mon, 25 Jan 2016 22:19:03 +0800
Subject: [PATCH 1033/5242] MLK-12262-2 ARM: imx: adjust LPDDR2 frequency
 scale flow for i.MX7D TO1.1

commit  01e3bfd41cd95905825a9618484eccb6636e753d from
https://source.codeaurora.org/external/imx/linux-imx.git

LPDDR2 frequency scale flow needs to be updated for i.MX7D
TO1.1 due to the CKE timing change.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/lpddr3_freq_imx.S |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/mach-imx/lpddr3_freq_imx.S b/arch/arm/mach-imx/lpddr3_freq_imx.S
index f9951c9..648d558 100644
--- a/arch/arm/mach-imx/lpddr3_freq_imx.S
+++ b/arch/arm/mach-imx/lpddr3_freq_imx.S
@@ -239,6 +239,20 @@
 	ldr	r7, =0x8
 	str	r7, [r5, #DDRPHY_OFFSETW_CON2]
 
+	/* LPDDR2 and LPDDR3 has different setting */
+	ldr	r8, [r4, #DDRC_MSTR]
+	ands	r8, r8, #0x4
+	beq	15f
+
+	ldr	r7, =0x08080808
+	str	r7, [r5, #DDRPHY_CA_DSKEW_CON0]
+	str	r7, [r5, #DDRPHY_CA_DSKEW_CON1]
+	ldr	r7, =0x0a0a0808
+	str	r7, [r5, #DDRPHY_CA_DSKEW_CON2]
+	ldr	r7, =0x0a0a0a0a
+	str	r7, [r5, #DDRPHY_CA_WLDSKEW_CON0]
+	b	14f
+15:
 	ldr	r7, [r9, #ANADIG_DIGPROG]
 	and	r7, r7, #0x11
 	cmp	r7, #0x10
-- 
1.7.9.5

